### PR TITLE
Fix chattr error when installing consul

### PR DESCRIPTION
**Pull request by shaidar**
When running the consul formula on a new instance, the following error is generated when running `unset_immutable_bit_on_resolv_conf` in the dns_proxy state:
`chattr: Operation not supported while reading flags on /etc/resolv.conf`

After reading about the problem, it appears that the issue can be resolved for both Debian and Ubuntu by installing the `resolvconf` pkg and modifying the `head` file

### CODE REVIEW

**Comment by blarghmatey** [comment]
Rather than investing more on hacking resolv.conf and using dnsmasq I think we should reassess our options based on https://learn.hashicorp.com/consul/security-networking/forwarding#systemd-resolved-setup

At this point we can safely use the systemd-resolved approach and either configure the ports for Consul or use iptables to reroute requests.

**Comment by shaidar** [comment]
Made changes based on the article referenced above and removed references in formula and tests for `dnsmasq` in addition to some minor cleanup.

File: consul/config.sls [hunk 1]
```diff
@@ -13,3 +13,26 @@ write_{{ name }}_config:
...
+configure_systemd_resolved_conf:
+  file.append:
+    - name: /etc/systemd/resolved.conf
+    - text: |
+        DNS=127.0.0.1
+        Domains=~consul
+
+{% for protocol in ['udp', 'tcp'] %}
+redirect_{{ protocol }}_dns_port_to_consul_port:
+  iptables.append:
```
Comment (blarghmatey) [inline consul/config.sls]: Just curious if you tried out changing the port number in Consul to avoid needing the IPTables redirect.

Comment (shaidar) [reply consul/config.sls]: U mean having consul listen on 53? No, I haven't tried it

Comment (blarghmatey) [reply consul/config.sls]: Yes, that way we don't even have to deal with iptables being in the loop.

Comment (shaidar) [reply consul/config.sls]: Removed iptables config and will add a separate PR for consul pillar to configure port.

File: consul/map.jinja [hunk 2]
```diff
@@ -10,8 +10,7 @@
         'service': 'consul',
         'conf_file': '/etc/consul/consul.conf',
         'config_dir': '/etc/consul.d',
-        'data_dir': '/var/lib/consul',
-        'dns_port': 8600
```
Comment (blarghmatey) [inline consul/map.jinja:14]: Might want to keep this and default it to port 53

Comment (shaidar) [reply consul/map.jinja:14]: It's not being used anywhere as the consul config file is generated from pillar data

Comment (blarghmatey) [reply consul/map.jinja:14]: Fair enough. I was just thinking we should maybe update the formula to provide that as a default.

**Comment by blarghmatey** [comment]
Closing this PR as it has drifted and is not required. We have decided to continue using dnsmasq rather than moving to iptables for DNS request routing.

**Pull request closed without merging**